packages/cf.js/src/app-factory.ts

```diff
@@ -1,5 +1,42 @@
-import { AppDefinition } from "./structs";
+import {
+  NodeMessageType,
+  NodeProposeInstallData
+} from "@counterfactual/node-provider";
+import { BigNumber, BigNumberish } from "ethers/utils";
+
+import { AppDefinition, BlockchainAsset } from "./app-types";
+import { Provider } from "./provider";
+import { Address, AppInstanceID } from "./simple-types";
+
+export interface InstallProposalRequest {
+  peerAddress: Address;
+  asset: BlockchainAsset;
+  myDeposit: BigNumberish;
+  peerDeposit: BigNumberish;
+  initialState: any;
+}
 
 export class AppFactory {
-  constructor(readonly appDefinition: AppDefinition) {}
+  constructor(
+    readonly provider: Provider,
+    readonly appDefinition: AppDefinition
+  ) {}
+
+  async proposeInstall(
+    proposal: InstallProposalRequest
+  ): Promise<AppInstanceID> {
+    const response = await this.provider.sendRawNodeRequest(
+      NodeMessageType.PROPOSE_INSTALL,
+      {
+        assetType: proposal.asset.assetType,
+        token: proposal.asset.token,
+        peerAddress: proposal.peerAddress,
+        myDeposit: new BigNumber(proposal.myDeposit).toString(),
+        peerDeposit: new BigNumber(proposal.peerDeposit).toString(),
```

Comment:
In `AppInstallProposal`, we define the deposits as `deposits: { [addr: string]: BigNumberish };`. Why not here, too?